examples/lines_bars_and_markers/masked_demo.py

```diff
@@ -1,30 +1,46 @@
-'''
-===========
-Masked Demo
-===========
+"""
+==============================
+Plotting masked and NaN values
+==============================
 
-Plot lines with points masked out.
+Sometimes you need to plot data with missing values.
 
-This would typically be used with gappy data, to
-break the line at the data gaps.
-'''
+One possibility is to remove the undesired points from the data. In this case,
+a plotted line throughout the data is contignuous. It connects all given data
+points.
+
+Alternatively, the undesired points can be removed either using a
+`masked array`_. No marker is drawn at points where *x* or *y* are masked. The
+line is broken there. It connects only contignuous ranges of non-masked data
+points.
+
+The same holds for NaN values. As far as plotting is concerned, masked values
+and NaN values are treated equivalently.
+
+.. _masked array:
+   https://docs.scipy.org/doc/numpy/reference/maskedarray.generic.html
+
+"""
 
 import matplotlib.pyplot as plt
 import numpy as np
 
-x = np.arange(0, 2*np.pi, 0.02)
-y = np.sin(x)
-y1 = np.sin(2*x)
-y2 = np.sin(3*x)
-ym1 = np.ma.masked_where(y1 > 0.5, y1)
-ym2 = np.ma.masked_where(y2 < -0.5, y2)
-
-lines = plt.plot(x, y, x, ym1, x, ym2, 'o')
-plt.setp(lines[0], linewidth=4)
-plt.setp(lines[1], linewidth=2)
-plt.setp(lines[2], markersize=10)
-
-plt.legend(('No mask', 'Masked if > 0.5', 'Masked if < -0.5'),
-           loc='upper right')
-plt.title('Masked line demo')
+
+x = np.linspace(-np.pi/2, np.pi/2, 31)
+y = np.cos(x)**3
+
```

Comment:
perhaps add a comment like "show various ways to remove the points where y>0.7" (approximately)?

Reply:
I've added a comment per case.